File: app/src/main/java/wmding/example/com/mylivetools/fragment/third/ThirdContract.java
package wmding.example.com.mylivetools.fragment.third;

import wmding.example.com.mylivetools.base.BasePresenter;
import wmding.example.com.mylivetools.base.BaseView;
import wmding.example.com.mylivetools.bean.CategoryResult;

/**
 * @author wmding
 * @date 2018/12/24
 * @describe
 */
public interface ThirdContract {

    interface Presenter extends BasePresenter {

        void getImages(boolean isRefresh, String category, int count, int page);
    }


    interface View extends BaseView<Presenter> {

        boolean isActive();

        void showImages(CategoryResult categoryResult);

        void addImages(CategoryResult categoryResult);

        /**
         * 设置下拉加载的控件
         *
         * @param isShow 是否显示
         */
        void setLoadingIndicator(boolean isShow);
    }
}

File: app/src/main/java/wmding/example/com/mylivetools/fragment/third/ThirdContract.kt
package wmding.example.com.mylivetools.fragment.third

import wmding.example.com.mylivetools.base.BasePresenter
import wmding.example.com.mylivetools.base.BaseView
import wmding.example.com.mylivetools.bean.CategoryResult

/**
 * @author wmding
 * @date 2018/12/24
 * @describe
 */
interface ThirdContract {

    interface Presenter : BasePresenter {

        fun getImages(isRefresh: Boolean, category: String, count: Int, page: Int)
    }


    interface View : BaseView<Presenter> {

        val isActive: Boolean

        fun showImages(categoryResult: CategoryResult)

        fun addImages(categoryResult: CategoryResult)

        /**
         * 设置下拉加载的控件
         *
         * @param isShow 是否显示
         */
        fun setLoadingIndicator(isShow: Boolean)
    }
}
